tox.ini

```diff
@@ -1,39 +1,46 @@
 [tox]
 envlist =
-    py{39,310,311}-test{,-oldestdeps,-devdeps,-predeps}{,-buildhtml}
+    py{39,310,311,312}-test{,-oldestdeps,-devdeps,-predeps}{,-buildhtml}
 requires =
     pip >= 19.3.1
 
 [testenv]
 
 description = run tests
 
+setenv =
+    devdeps: PIP_EXTRA_INDEX_URL =  https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
+
 deps =
     # We use these files to specify all the dependencies, and below we override
     # versions for specific testing schenarios
     -rtest_requirements.txt
     -rsite/requirements.txt
     -rrequirements.txt
 
-    oldestdeps: numpy==1.20
-    oldestdeps: matplotlib==3.4
-    oldestdeps: scipy==1.6
-    oldestdeps: pandas==1.2
-    oldestdeps: statsmodels==0.13
+    oldestdeps: numpy==1.23
+    oldestdeps: matplotlib==3.6
+    oldestdeps: scipy==1.8
+    oldestdeps: pandas==1.4
+    oldestdeps: statsmodels==0.13.1
+
+    devdeps: numpy>=0.0.dev0
+    devdeps: scipy>=0.0.dev0
+    devdeps: matplotlib>=0.0.dev0
+    devdeps: pandas>=0.0.dev0
+    devdeps: statsmodels>=0.0.dev0
 
 allowlist_externals = bash
 
 commands =
-    devdeps: pip install -U --pre --only-binary :all: -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
```

Comment:
I don't know how to properly specify this in tox (i.e. these lines are being removed), but I think we'd also want to add the `--no-deps` flag in order to prevent versions being downgraded.

Reply:
Commands come after the other dependencies are installed, so it was an effective way to override the downgraded install. But I'll add the flag, too.

Long term will need some version number check as the job happily returned a passing green status even when the versions were downgraded and haven't been the ones required by the config.